INSTALL.md

```diff
@@ -77,3 +77,16 @@ Please see [INSTALL.make.md](dev/doc/INSTALL.make.md) for build and
 installation instructions using `make`. If you wish to experiment with
 the Dune-based system see the [dune guide for
 developers](dev/doc/build-system.dune.md).
+
+Run-time dependencies of native compilation
+-------------------------------------------
+
+The OCaml compiler and findlib are build-time dependencies, but also
+run-time dependencies if you wish to use the native compiler.
+
+OCaml toolchain advisory
+------------------------
+
+When loading plugins or `vo` files, you should make sure that these
+were compiled with the same OCaml setup (version, flags,
+dependencies...) as Coq.
```

Comment:
It just occurred to me a couple of further comments [sorry for the churn]
- we don't mention the actual error anywhere it seems. So users may have trouble mapping the dreaded "implementation mismatch on Foo" to this error.
- should something be also added to reference manual?

Regarding your comment on distribution, I dunno. Maybe I would phrase it positively "distribution of pre-compiled plugins and `.vo` files is only possible if users are guaranteed to have the exact same OCaml toolchain".